src/application.rs

```diff
@@ -286,13 +285,13 @@ pub trait ApplicationHandler {
     /// with the [`onStop`] lifecycle event which typically results in the surface to be destroyed
     /// after the app becomes invisible.
     ///
-    /// Applications that need to run on Android should assume their [`NativeWindow`] has been
-    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
-    /// created outside of Winit (such as an `EGLSurface`, [`VkSurfaceKHR`] or [`wgpu::Surface`]).
+    /// Applications that need to run on Android must be able to handle their underlying
+    /// [`SurfaceView`] being destroyed, which in turn indirectly invalidates any existing
+    /// render surfaces that may have been created outside of Winit (such as an `EGLSurface`,
+    /// [`VkSurfaceKHR`] or [`wgpu::Surface`]).
     ///
-    /// When receiving [`destroy_surfaces()`] Android applications should drop all render surfaces
-    /// before the event callback completes, which may be re-created when the application next
-    /// receives [`can_create_surfaces()`].
+    /// This means that in this method, you must drop all render surfaces before the event callback
+    /// completes, and only re-create them in or after [`can_create_surfaces()`] is next recieved.
```

Comment:
This looks to be accidentally reverting the documentation fixes from #3786, which were initially discovered in #3505. Perhaps this was due to a conflict-resolve?

Reply:
Oops, you're right!